ssm_cache/cache.py

```diff
@@ -5,15 +5,27 @@
 from functools import wraps
 import six
 
-import boto3
-
 class InvalidParameterError(Exception):
     """ Raised when something's wrong with the provided param name """
 
 class Refreshable(object):
     """ Abstract class for refreshable objects (with max-age) """
 
-    ssm_client = boto3.client('ssm')
+    _ssm_client = None
+
+    @classmethod
+    def set_ssm_client(cls, client):
+        """Override the default boto3 SSM client with your own."""
+        if not 'get_parameters' in dir(client):
```

Comment:
This is a good check for now, but it could easily get out of hands if we start using more methods/APIs. I think it's ok for now.

Reply:
It's not something I would typically do, but included because of your request to fail early in previous reviews on this branch. It seems to me that this would prevent client implementations that used dynamic code generation, but as you said it seems OK for now.

Reply:
It made more sense for our previous client/session factory because we wanted them to have exactly that method. On the other hand, a boto3 client has plenty of methods, and we might use more than one or two in the future. But no worries for now.